docs/examples/external-dns-rbac.yaml

```diff
@@ -0,0 +1,61 @@
+apiVersion: v1
```

Comment:
Hi, i'd suggest we separate this into two file. rbac.yaml and deployment.yaml.
So we don't need to maintain two separate external-dns deployment configuration.
Maybe inside a separate folder like docs/example/external-dns/rbac.yaml  and docs/example/external-dns/deployment.yaml ?

Reply:
The deployments are not 100% identical.   RBAC sets serviceAccountName: external-dns
Solution: How about only including the newer RBAC version now?  Add a comment "for non-RBAC deployments refer to the external-dns tutorial", where that code is still available.

Reply:
agree, let's delete the non-rbac version 👍

Reply:
updated